dateparser/search/search.py

```diff
@@ -86,10 +86,15 @@ def split_if_not_parsed(self, item, original):
                 possible_splits.extend(self.split_by(item, original, splitter))
         return possible_splits
 
-    def parse_item(self, parser, item, translated_item, parsed, need_relative_base):
+    def parse_item(self, parser, item, translated_item, parsed, need_relative_base, language):
         relative_base = None
-        item = item.replace('ngày', '')
-        item = item.replace('am', '')
+
+        if language != "en":
+            item = item.replace('ngày', '')
+            item = item.replace('am', '')
```

Comment:
why should we apply this only in English? There's a lot of languages that use `am` and `pm`.

Reply:
Surething, even Vietnamese is not causing the issue it's actually German.

ie,

`am 8 ` means `on the 8th`

but 

`8 am` actually means `8 am`

so however it is done it presents problem one way or other.

Reply:
@gavishpoddar, it makes totally sense, however, the next time someone would arrive here wouldn't know the reason, so I think you could explain this in a comment :)